client/middleware/asyncMiddleware.js

```diff
@@ -0,0 +1,22 @@
+import * as actionTypes from '../actions/actionTypes';
+import * as actions from '../actions/actions'
+
+const asyncMiddleware = store => next => action => {
+    // Run this middleware for every action that requires async
+    if(action.async) {
+        action.promise
+        .then(value => {
+            store.dispatch(value);
+        });
+    } else {
+        return next(action);
+    }
+}
+
+export const asyncAction = (promise) => ({
+    type: actionTypes.PROMISE,
+    async: true,
+    promise
+});
+
+export default asyncMiddleware;
```

Comment:
This is the `asyncMiddleware` there's a lot going on here. Basically, whenever an `asyncAction` (with a promise attached to it) is dispatched, it triggers the middleware which evaluates the promise. It then returns the next action. Essentially, this middleware is necessary to teach Redux how to work asynchronously. There are middlewares out there that do this (`redux-thunk` is an example) but I decided to make my own

Reply:
But you are using redux-thunk.